### PR TITLE
Switch gcs2cbt to use MutationBatcher.

### PR DESCRIPTION
**Pull request by dopiera**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2117)

### CODE REVIEW

**Comment by coryan** [comment]
Woo hoo! :tada: :tada: :tada:

File: google/cloud/examples/gcs2cbt.cc [hunk 1]
```diff
@@ -193,33 +137,11 @@ int main(int argc, char* argv[]) try {
...
-  };
-
   std::cout << "Starting " << thread_pool_size << " workers ..." << std::flush;
-  std::vector<std::future<void>> workers;
-  for (std::size_t i = 0; i != thread_pool_size; ++i) {
-    workers.push_back(std::async(std::launch::async, read_buffer, table,
-                                 report_worker_progress_rate));
+  google::cloud::bigtable::CompletionQueue cq;
+  std::vector<std::thread> thread_pool;
+  for (int i = 0; i < static_cast<int>(thread_pool_size); ++i) {
```
Comment (devjgm) [inline google/cloud/examples/gcs2cbt.cc]: I suggest declaring `i` to be std::size_t and avoid the cast.

Comment (dopiera) [reply google/cloud/examples/gcs2cbt.cc]: Done

**Comment by coryan** [comment]
___
*[google/cloud/examples/gcs2cbt.cc, line 143 at r4](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2117#-LZp2HvR4FEhG1oipWzG:-LZp2HvR4FEhG1oipWzH:b-myt9xj) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/a4d2b2ca0792c3d67f8e302f6270f9c5451bde53/google/cloud/examples/gcs2cbt.cc#L143)):*
> ```cpp
>   google::cloud::bigtable::CompletionQueue cq;
>   std::vector<std::thread> thread_pool;
>   for (size_t i = 0; i < thread_pool_size; ++i) {
> ```

nit: s/size_t/std::size_t/

**Pull request merged**